### ISSUE TITLE
Test output order has changed: test result shown after `afterEach` hook output

### ISSUE THREAD

**Issue by juergba** [mochajs/mocha#3569]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend avoiding the use of globally installed Mocha.

### Description

The output of the testfile [cascade.fixture.js](https://github.com/mochajs/mocha/blob/master/test/integration/fixtures/cascade.fixture.js) has changed compared to my local V5.2.0. The order of hook execution remains correct.

Output of master branch:
```
one
before one
    two
  before two
      three
    before three
  before each one
    before each two
    before each three
      TEST three
    after each three
    after each two
  after each one
        √ should three                            <<<<< different
    after three
  after two
after one
```
Output of my local Mocha installation V5.2.0:
```
one
before one
    two
  before two
      three
    before three
  before each one
    before each two
    before each three
      TEST three
        √ should three
    after each three
    after each two
  after each one
    after three
  after two
after one
```

### Steps to Reproduce

**Expected behavior:** as shown in second output

**Actual behavior:** as shown in first output

**Reproduces how often:** always

### Versions

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`: 5.2.0
- The output of `node --version`: 10.5.0
- The version and architecture of your operating system: Windows 10 Home, 1809
- Your shell (bash, zsh, PowerShell, cmd, etc.): cmd of VSCode
- Your browser and version (if running browser tests): 
- Any other third party Mocha related modules (with versions):
- The code transpiler being used: none

**Comment by juergba** [comment]
```
D:\mocha3291\mocha>git show --oneline -s
593e9b8 (HEAD -> master, origin/master, origin/HEAD) Add ability to query Mocha version programmatically (#3535)
```

**Comment by plroebuck** [comment]
I glanced through the commits but nothing popped out. We'll have to create a test that verifies the _exact_ expected order, then checkout commits since 5.2.0 (jumping by half), add the test manually, and retest until one of them fails.

**Comment by juergba** [comment]
I checked-out the commits:
- 02294a3: 11. August - fix report in a case of afterEach conditional fail
- 2bb4124: 07. August - resolves #3450: changed links labeled "code of conduct" from changelo

The first commit fails, while using above testfile.
